### PR TITLE
Prevent the creation of object IDs that look like views

### PR DESCRIPTION
**Pull request by dataflake**
The change disallows objects with IDs that start with either `@@` or `++` so users cannot create something that will then be unreachable because Zope tries to look up a view with that name.

### CODE REVIEW

**Comment by dataflake** [comment]
Fixes #591

File: src/OFS/ObjectManager.py [hunk 1]
```diff
@@ -125,6 +125,10 @@ def checkValidId(self, id, allow_dup=0):
         raise BadRequest(
             'The id "%s" is invalid because it '
             'ends with two underscores.' % id)
+    if id.startswith('@@') or id.startswith('++'):
+        raise BadRequest(
+            'The id "%s" is invalid because it starts with characters '
+            'reserved for Zope views lookup.' % id)
```
Comment (icemac) [inline src/OFS/ObjectManager.py]:
```suggestion
            "reserved for Zope's view lookup." % id)
```

Comment (dataflake) [reply src/OFS/ObjectManager.py]: Das ist eine Verschlimmbesserung, geht nicht.

Comment (icemac) [reply src/OFS/ObjectManager.py]: Sorry, maybe I did not understand the intent of the sentence of the exception.

Comment (dataflake) [reply src/OFS/ObjectManager.py]: It's a matter of better English

**Pull request merged**